src/Samples/NewWindow.Core/Program.fs

```diff
@@ -11,22 +11,22 @@ open Elmish.WPF
 open AppModule
 
 
-let main mainWindow (createWindow1: Func<#Window>) (createWindow2: Func<#Window>) =
+let main (mainWindow: Window) (createWindow1: Func<#Window>) (createWindow2: Func<#Window>) : unit =
   let logger =
     LoggerConfiguration()
       .MinimumLevel.Override("Elmish.WPF.Update", Events.LogEventLevel.Verbose)
       .MinimumLevel.Override("Elmish.WPF.Bindings", Events.LogEventLevel.Verbose)
       .MinimumLevel.Override("Elmish.WPF.Performance", Events.LogEventLevel.Verbose)
       .WriteTo.Console()
       .CreateLogger()
-  let createWindow1 () = createWindow1.Invoke()
-  let createWindow2 () =
+  let createWindow1 unit : #Window = createWindow1.Invoke()
```

Comment:
I don't think this is a refactor.  (I can't check because I don't have my personal computer with me now.)

Before, the argument had to be `()`, which is the only value of type `Unit`.  Instead, by giving this parameter a name, you are allowing F# to infer the type.  But this parameter is unused, so I think F# infers its type is `obj`.

Same problem in other places.

Reply:
I had thought I'd read in documentation that `unit` and `()` were synonymous.  I could very well be mistaken.  

See https://web.archive.org/web/20221225214301/https://fsharpforfunandprofit.com/posts/how-types-work-with-functions/
... a little less than 1/2 down Scott W. says...
"
What is this “unit”?

Well, even if a function returns no output, it still needs a range. There are no “void” functions in mathematics-land. Every function must have some output, because a function is a mapping, and a mapping has to have something to map to!

So in F#, functions like this return a special range called “unit”. This range has exactly one value in it, called “()”. You can think of unit and () as somewhat like “void” (the type) and “null” (the value) in C#. But unlike void/null, unit is a real type and () is a real value.
"

As I said I'm stuck with what I read but you know F# better than I.  

I do know that when I read `()` I found it less clearly distinct from all the other uses of paretheses than the reserved word `unit` so where possible I used `unit`.

Reply:
Tested on my machine.

```fsharp
let foo = unit // The value or constructor 'unit' is not defined.
```

`unit` is not a keyword in F#. You just named an argument of unspecified type `unit`.

Repeating because it's relevant here: `unit` looks the way it does so that parameterless function calls in F# can resemble parameterless function calls in other languages. Parameterless function `python_func` in Python will be called as `python_func()`, and parameterless function `fSharpFunc` in F# will be called as `fSharpFunc ()`. The creators of F# intended the language to be used this way and everyone (of whom I am aware) uses it this way.

Reply:
Looks as though it's generic (`'a`), so you could pass, say, an integer in if you wanted, which would mean someone was doing something wrong. Further, it's detected as unused now, which annotating its type does not resolve (and `unit: unit` looks ugly anyway).

Reply:
I _think_ I removed the place where `unit : unit` appeared.